optics-core/src/Data/Tuple/Optics.hs

```diff
@@ -46,78 +50,91 @@ import Data.Functor.Product
 import Data.Kind
 import Data.Proxy
 import GHC.Generics ((:*:)(..), Generic(..), K1, M1, U1)
+import GHC.TypeLits (Nat, type (-))
 
 import GHC.Generics.Optics
+import Optics.Iso
 import Optics.Lens
 import Optics.Optic
 
+class PositionOptic (pos :: Nat) k s t a b | pos s -> k a
+                                           , pos t -> k b
+                                           , pos s b -> t
+                                           , pos t a -> s where
+    positionOptic :: Optic k NoIx s t a b
+    default positionOptic :: (k ~ A_Lens, Generic s, Generic t, GIxed (NatToN pos) (Rep s) (Rep t) a b) => Optic k NoIx s t a b
+    positionOptic = ix (Proxy :: Proxy (NatToN pos))
+    {-# INLINE [1] positionOptic #-}
+
+type family NatToN (n :: Nat) :: Type where
+    NatToN 0 = Z
+    NatToN n = S (NatToN (n - 1))
+
 -- | Provides access to 1st field of a tuple.
-class Field1 s t a b | s -> a, t -> b, s b -> t, t a -> s where
-  -- | Access the 1st field of a tuple (and possibly change its type).
-  --
-  -- >>> (1,2) ^. _1
-  -- 1
-  --
-  -- >>> (1,2) & _1 .~ "hello"
-  -- ("hello",2)
-  --
-  -- >>> traverseOf _1 putStrLn ("hello","world")
-  -- hello
-  -- ((),"world")
-  --
-  -- This can also be used on larger tuples as well:
-  --
-  -- >>> (1,2,3,4,5) & _1 %~ (+41)
-  -- (42,2,3,4,5)
-  _1 :: Lens s t a b
-  default _1 :: (Generic s, Generic t, GIxed N0 (Rep s) (Rep t) a b)
-             => Lens s t a b
-  _1 = ix proxyN0
-  {-# INLINE[1] _1 #-}
+type Field1 = PositionOptic 0 A_Lens
```

Comment:
I wonder about the impact on type error messages. Presumably this type synonym will tend to get expanded, and the off-by-one is mildly confusing.